### PR TITLE
fix(gatsyby): improve error messages for errors outside of react"

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Updated the correct error message in develop so people know where to look

Before:
image

After:
image

## Related Issues

Fixes https://github.com/gatsbyjs/gatsby/issues/30046

[ch26500]

### CODE REVIEW

File: packages/gatsby/cache-dir/__tests__/ensure-resources.tsx [hunk 1]
```diff
@@ -4,6 +4,9 @@ import { render, getNodeText, cleanup } from "@testing-library/react"
 
 jest.mock(`../loader`, () => {
```
Comment (wardpeet) [inline packages/gatsby/cache-dir/__tests__/ensure-resources.tsx:5]: This is a really dumb test but :shrug:

File: packages/gatsby/cache-dir/loader.js [hunk 2]
```diff
@@ -348,12 +348,20 @@ export class BaseLoader {
...
+      const pageData = this.pageDb.get(pagePath)
+
+      if (pageData.payload) {
+        return pageData.payload
+      }
+
+      return {
+        error: pageData.error,
+        status: pageData.status,
+      }
```
Comment (wardpeet) [inline packages/gatsby/cache-dir/loader.js:364]: This is a behavior change.

Comment (pieh) [reply packages/gatsby/cache-dir/loader.js:364]: I think change itself make sense but we have couple of places where we call this other than `ensure-resources` - including exposing it in public API ( https://github.com/gatsbyjs/gatsby/blob/ec57fb2e8b92c99db991e226383cfd7642eb511f/packages/gatsby/cache-dir/api-runner-browser.js#L27-L29 ), but also other places in our runtime - would it make sense to make it optional to behave as propsoed here - i.e. `loadPageSync(rawPath, { getErrorDetailsInsteadOfUndefined: true })`?

Comment (wardpeet) [reply packages/gatsby/cache-dir/loader.js:364]: I was thinking of that too but wasn't sure if it's worth it :shrug:

File: packages/gatsby/cache-dir/loader.js [hunk 3]
```diff
@@ -546,7 +556,9 @@ export const publicLoader = {
   getResourceURLsForPathname: rawPath =>
     instance.getResourceURLsForPathname(rawPath),
   loadPage: rawPath => instance.loadPage(rawPath),
-  loadPageSync: rawPath => instance.loadPageSync(rawPath),
+  // TODO add deprecation to v4 so people use withErrorDetails and than we can remove in v5 and change default behviour
```
Comment (wardpeet) [inline packages/gatsby/cache-dir/loader.js:559]:
```suggestion
  // TODO add deprecation to v4 so people use withErrorDetails and then we can remove in v5 and change default behaviour
```

**Comment by bradlocking** [comment]
Any idea when we might see this land as a version bump?

**Review by pieh — APPROVED** [review]

**Pull request merged**